simulator/src/main/java/com/hedera/block/simulator/grpc/impl/ConsumerStreamObserver.java

```diff
@@ -70,6 +75,9 @@ public ConsumerStreamObserver(
     @Override
     public void onNext(SubscribeStreamResponse subscribeStreamResponse) {
         final SubscribeStreamResponse.ResponseCase responseType = subscribeStreamResponse.getResponseCase();
+        if (lastKnownStatuses.size() == lastKnownStatusesCapacity) {
```

Comment:
I think better do >= on line 78, since it is way less error prone. There could be a case where we could go above the capacity before we make this check, which means we will never enter the `if`. I do not think we need to be _exact_ here for the simulator, but rather clean the collection as long as we are overboard. Also, that being said, if we are ever above the capacity, maybe we should poll the difference, as big as it is, in this way we ensure we always keep proper capacity.

Reply:
Very good point, thanks 👍